pkg/provider/azure_controller_common.go

```diff
@@ -142,6 +142,13 @@ func (c *controllerCommon) getNodeVMSet(nodeName types.NodeName, crt azcache.Azu
 	}
 
 	// 4. Node is managed by vmss
+	if _, err := ss.GetInstanceIDByNodeName(mapNodeNameToVMName(nodeName)); err != nil {
```

Comment:
We should invoke `vmas.GetInstanceIDByNodeName` in `ss.GetInstanceIDByNodeName` instead of modifying the `vmSet` directly. This is because it is recommended to set the vmSet type to vmss when using standalone VMs.